exporter/awskinesisexporter/config.go

```diff
@@ -14,6 +14,7 @@ import (
 // AWSConfig contains AWS specific configuration such as awskinesis stream, region, etc.
 type AWSConfig struct {
 	StreamName      string `mapstructure:"stream_name"`
+	StreamARN       string `mapstructure:"stream_arn"`
```

Comment:
I was also considering of adding validations, but decided to leave the validation to AWS api, since stream_name also has rule of the length and pattern.

>Length Constraints: Minimum length of 1. Maximum length of 128.
Pattern: [a-zA-Z0-9_.-]+

But different from the ARN, AWS api will return the clear error message with invalid stream name:
>at 'streamName' failed to satisfy constraint: Member must satisfy regular expression pattern: [a-zA-Z0-9_.-]+

And the ARN only shows:
>The input ARN "arn:aws:kinesis:xxx:0:stream/invalid-stream$" is invalid

Base on the validation styles in Opentelemetry collector projects, do you think we should do the ARN validation? 
The implementation is clear and easy to add: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2@v1.30.4/aws/arn#IsARN

---
reference: https://docs.aws.amazon.com/kinesis/latest/APIReference/API_DescribeStream.html#API_DescribeStream_RequestSyntax